test/support/wait-for.ts

```diff
@@ -0,0 +1,7 @@
+import { TIME_EXCEEDING_ACBS_AUTHENTICATION_ID_TOKEN_CACHE_TTL } from './environment-variables';
+
+// TODO APIM-97: should we use fake timers instead?
+// TODO APIM-97: should all tests wait for cache to expire?
```

Comment:
In the current implementation, only the caching API tests and the API tests that check the error cases for getting id tokens wait for the cache to expire. Our other API tests don't wait for the cache to expire. The upside is that it's simpler and quicker this way, and it means that the caching behaviour gets tested naturally. The downside is that it couples tests together slightly which might be surprising for someone in future / might cause a surprising bug, and it means that we don't _explicitly_ test caching in each endpoint.

I think the simplicity and speed of this implementation is worth the downsides, but I wanted to highlight this in case anyone disagreed

Reply:
I like this testing approach, but my experience is limited here. Less repeating tests and maybe other API tests doesn't need to know about caching.